juju/juju
# LXD availability zones

## Description of change

This patch adds initial support for "zone" placement directives in the LXD provider.

The notion of a zone is expected to evolve in the upstream LXD provider, but the current concept (and the one implemented here) is for a zone to be synonymous with a server name from the current list of LXD cluster members.

Supported syntax is "zone={zone}" or just "{zone}".

## QA steps

Supposing a LXD cluster cloud "lxd-guimaas" with nodes "nuc5" and "nuc6":
- `juju bootstrap lxd-guimaas zone-test --no-gui --debug --to zone=nuc5` and check that the controller is created on nuc5
- `juju deploy ubuntu --to zone=nuc6` and check that a new container is created on nuc6.
- `juju add-unit ubuntu --to nuc6` multiple times and check that a new container is always created on nuc6.

## Documentation changes

Yes.

## Bug reference

N/A
